pimd/pim_cmd_common.c

```diff
@@ -428,23 +428,27 @@ void pim_show_neighbors_secondary(struct pim_instance *pim, struct vty *vty)
 
 		for (ALL_LIST_ELEMENTS_RO(pim_ifp->pim_neighbor_list, neighnode,
 					  neigh)) {
-			char neigh_src_str[INET_ADDRSTRLEN];
 			struct listnode *prefix_node;
 			struct prefix *p;
 
 			if (!neigh->prefix_list)
 				continue;
 
-			pim_inet4_dump("<src?>", neigh->source_addr,
-				       neigh_src_str, sizeof(neigh_src_str));
-
 			for (ALL_LIST_ELEMENTS_RO(neigh->prefix_list,
 						  prefix_node, p))
-				vty_out(vty, "%-16s %-15s %-15s %-15pFX\n",
+#if PIM_IPV == 4
+				vty_out(vty, "%-16s %-15s %-15pPAs %-15pFX\n",
 					ifp->name,
 					inet_ntop(AF_INET, &ifaddr, buf,
 						  sizeof(buf)),
-					neigh_src_str, p);
+					&neigh->source_addr, p);
+#else
+				vty_out(vty, "%-16s %-15s %-15pPAs %-15pFX\n",
+					ifp->name,
+					inet_ntop(AF_INET6, &ifaddr, buf,
```

Comment:
you need to change ifaddr type from struct in_addr to pim_addr and then use %pPAs here as well. No need of #if and #else